pkg/calculator/adder_test.go

```diff
@@ -21,10 +21,11 @@ func TestAdd(t *testing.T) {
 	}, {
 		100, 200, 300,
 	}}
-	for _, arg := range args {
-		t.Run(fmt.Sprintf("%d+%d=%d", arg.i, arg.j, arg.expected), func(t *testing.T) {
+	for _, tt := range tests {
+		tt := tt // Capture range variable.
+		t.Run(fmt.Sprintf("%d+%d=%d", tt.i, tt.j, tt.expected), func(t *testing.T) {
```

Comment:
IMHO shadowing variables, even deliberately, is not a good style because it's harder to reason about and may lead to subsequent bugs being introduced. Therefore I usually follow the pattern of wrapping the consuming code in a func and passing the to-be-shadowed variable as a parameter:
```suggestion
		func(tt test) {
			t.Run(fmt.Sprintf("%d+%d=%d", tt.i, tt.j, tt.expected), func(t *testing.T) {
				t.Parallel()
				assert.Equal(t, tt.expected, calculator.Add(tt.i, tt.j))
			})
		}(tt)
```

Reply:
This is also shadowing... yet with more complexity. It seems pretty idiomatic to capture range variable like the original revision in the PR.

Reply:
https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing also uses the shadow variable in range loop btw.

Reply:
Ok, as the Golang wiki prescribes this way of doing it, we should just follow that lead. I still don't like it, though.